app/src/main/java/com/x8bit/bitwarden/data/platform/manager/restriction/RestrictionManagerImpl.kt

```diff
@@ -0,0 +1,131 @@
+    private var isReceiverRegistered = false
+
+    init {
+        appForegroundManager
+            .appForegroundStateFlow
+            .onEach {
+                when (it) {
+                    AppForegroundState.BACKGROUNDED -> handleBackground()
+                    AppForegroundState.FOREGROUNDED -> handleForeground()
+                }
+            }
+            .launchIn(mainScope)
+    }
+
+    override fun updatePreconfiguredRestrictionSettings() {
+        restrictionsManager
+            .applicationRestrictions
+            ?.takeUnless { it.isEmpty }
+            ?.let { setPreconfiguredSettings(it) }
+    }
+
+    private fun handleBackground() {
+        if (isReceiverRegistered) {
+            context.unregisterReceiver(restrictionsChangedReceiver)
+        }
+        isReceiverRegistered = false
+    }
+
+    private fun handleForeground() {
+        context.registerReceiver(restrictionsChangedReceiver, intentFilter)
+        isReceiverRegistered = true
+        updatePreconfiguredRestrictionSettings()
+    }
+
+    private fun setPreconfiguredSettings(bundle: Bundle) {
+        bundle
+            .getString(BASE_ENVIRONMENT_URL_RESTRICTION_KEY)
+            ?.let { url -> setPreconfiguredUrl(baseEnvironmentUrl = url) }
+    }
+
+    private fun setPreconfiguredUrl(baseEnvironmentUrl: String) {
+        environmentRepository.environment = when (val current = environmentRepository.environment) {
+            Environment.Us -> {
+                when (baseEnvironmentUrl) {
+                    // If the base matches the predefined Us environment, leave it alone
+                    Environment.Us.environmentUrlData.base -> current
+                    // If the base does not match the predefined Us environment, create a
+                    // self-hosted environment with the new base
+                    else -> current.toSelfHosted(base = baseEnvironmentUrl)
+                }
+            }
+
+            Environment.Eu -> {
+                when (baseEnvironmentUrl) {
+                    // If the base matches the predefined Eu environment, leave it alone
+                    Environment.Eu.environmentUrlData.base -> current
+                    // If the base does not match the predefined Eu environment, create a
+                    // self-hosted environment with the new base
+                    else -> current.toSelfHosted(base = baseEnvironmentUrl)
+                }
+            }
+
+            is Environment.SelfHosted -> {
+                when (baseEnvironmentUrl) {
+                    // If the base matches the self-hosted environment, leave it alone
+                    current.environmentUrlData.base -> current
+                    // If the base does not match the self-hosted Eu environment, create a
+                    // new self-hosted environment with the new base
+                    else -> current.toSelfHosted(base = baseEnvironmentUrl)
+                }
```

Comment:
I don't think we need the extra logic here right? Once it's already self hosted, can't this just be

```kotlin
is Environment.SelfHosted -> {
    current. toSelfHosted(base = baseEnvironmentUrl)
}
```

Reply:
Yea, we can simplify this